Source/SwiftLintFramework/Rules/Style/OptionalEnumCaseMatchingRule.swift

```diff
@@ -133,26 +164,61 @@ public struct OptionalEnumCaseMatchingRule: SubstitutionCorrectableASTRule, Conf
-            .compactMap { dictionary in
+            .flatMap { dictionary -> [NSRange] in
                 guard let offset = dictionary.offset, let length = dictionary.length else {
-                    return nil
+                    return []
                 }
 
-                let tokens = file.syntaxMap
-                    .tokens(inByteRange: NSRange(location: offset, length: length))
-                    .prefix(while: { $0.kind != .keyword })
+                let pattern = contents.substringWithByteRange(start: offset, length: length)
+                let tupleCommaByteOffsets = pattern?.tupleCommaByteOffsets ?? []
 
-                guard let lastToken = tokens.last else {
-                    return nil
+                let tokensToCheck = (tupleCommaByteOffsets + [length]).compactMap { length in
+                    return file.syntaxMap
+                        .tokens(inByteRange: NSRange(location: offset, length: length))
+                        .prefix { $0.kind != .keyword || file.isTokenUnderscoreKeyword($0) }
+                        .last
                 }
 
-                let questionMarkByteOffset = lastToken.length + lastToken.offset
-                guard contents.substringWithByteRange(start: questionMarkByteOffset, length: 1) == "?",
-                    let range = contents.byteRangeToNSRange(start: questionMarkByteOffset, length: 1) else {
-                    return nil
+                return tokensToCheck.compactMap { tokenToCheck in
+                    let questionMarkByteOffset = tokenToCheck.length + tokenToCheck.offset
+                    guard contents.substringWithByteRange(start: questionMarkByteOffset, length: 1) == "?" else {
+                        return nil
+                    }
+                    return contents.byteRangeToNSRange(start: questionMarkByteOffset, length: 1)
                 }
-
-                return range
             }
     }
 }
+
+private extension String {
+    func ranges(of substring: String) -> [Range<Index>] {
+        var ranges = [Range<Index>]()
+        while let range = range(of: substring, range: (ranges.last?.upperBound ?? startIndex)..<endIndex) {
+            ranges.append(range)
+        }
+        return ranges
+    }
+
+    var isTuple: Bool {
+        return first == "(" && last == ")" && contains(",")
+    }
+
+    var tupleCommaByteOffsets: [Int] {
+        guard isTuple else {
+            return []
+        }
+
+        let stringView = StringView(self)
+        return ranges(of: ",").map { range in
+            return stringView.byteOffset(fromLocation: distance(from: startIndex, to: range.lowerBound))
+        }
+    }
+}
+
+private extension SwiftLintFile {
+    func isTokenUnderscoreKeyword(_ token: SwiftLintSyntaxToken) -> Bool {
+        return token.kind == .keyword &&
+            token.length == 1 &&
+            stringView.substringWithByteRange(start: token.offset, length: token.length) == "_"
```

Comment:
IIRC we have an utility method somewhere that returns a string for `SwiftLintSyntaxToken`

Reply:
Found it.